lib/Doctrine/DBAL/Driver/PDOSqlsrv/Driver.php

```diff
@@ -61,16 +64,65 @@ private function _constructPdoDsn(array $params)
         }
 
         if (isset($params['dbname'])) {
-            $dsn .= ';Database=' .  $params['dbname'];
+            $dsn .= ';Database=' . $params['dbname'];
         }
 
-        if (isset($params['MultipleActiveResultSets'])) {
-            $dsn .= '; MultipleActiveResultSets=' . ($params['MultipleActiveResultSets'] ? 'true' : 'false');
+        foreach ($this->getDsnParams($params) as $dsnParamName => $dsnParamValue) {
+            $dsn .= sprintf(';%s=%s', $dsnParamName, $dsnParamValue);
         }
 
         return $dsn;
     }
 
+    /**
+     * Returns present dsn params
+     *
+     * @param string[] $params
+     *
+     * @return string[] Present DSN params
+     */
+    private function getDsnParams(array $params)
+    {
+        $dsnParams          = [];
+        $availableDsnParams = $this->getAvailableDsnParam();
+        $paramsLowercase    = array_change_key_case($params);
+
+        foreach ($availableDsnParams as $availableDsnParamName) {
+            $availableDsnParamValue = $paramsLowercase[strtolower($availableDsnParamName)] ?? null;
+            if (! isset($availableDsnParamValue)) {
+                continue;
+            }
+            $dsnParams[$availableDsnParamName] = $availableDsnParamValue;
+        }
+        return $dsnParams;
+    }
+
+    /**
+     * Returns available DSN params for driver
+     *
+     * @return string[] Available DSN params
+     */
+    private function getAvailableDsnParam()
+    {
+        return [
+            'APP',
```

Comment:
https://pecl.php.net/package-changelog.php?package=sqlsrv&release=5.2.0 introduces the `ColumnEncryption` parameter which indicates that it's better if we don't maintain this list ourselves at all.